charts/rancher-monitoring/v0.0.8/charts/prometheus/templates/prometheus.yaml

```diff
@@ -151,10 +151,13 @@ spec:
   serviceMonitorSelector:
 {{ toYaml .Values.serviceMonitorSelector | indent 4 }}
 {{- end }}
-{{- if and (eq .Values.level "project") (eq .Values.sync.mode "remote") }}
+{{- if .Values.remoteRead }}
   remoteRead:
-  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
-    url: {{ printf "%s%s" .Values.sync.target .Values.sync.path }}
+{{ toYaml .Values.remoteRead | indent 4 }}
+{{- end }}
```

Comment:
For the read  / write specs its a list of map. So the UX would be adding:

```
prometheus.remoteRead[0].url = http://remote
prometheus.remoteRead[0].bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
```
and similarly for remoteWrite?
```

Reply:
Hi @cloudnautique 

The question is what fields we would like to expose for remote read and remote write in Rancher UI. 

We only exposed remote write url in the UI PR. Other fields such as `bearerTokenFile` need to be configured in custom answers section. 

We can do the same for remote read. Is that OK if we only expose URL field?

https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#remotewritespec
https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#remotereadspec

Reply:
My question is what would the UX look like. My fear is the user will need to keep tabs on the index... 

In the UI if they use:
```
url1: https://x.y
url2: https://z.y
```

then they would need to know that
`prometheus.remoteRead[0]` applies to the first url.

It is kind of ugly. If we needed to do that then, I would want to see the index number in the UI.

Reply:
@loganhz, I do not want the UI if it only shows the URL box. Our experience has been that TLS configs and basic auth are going to be more common when it comes to these things. Having to use two separate input methods feels wrong.

I would accept a UI field that took a complete remote{read|write}spec or just use the advanced options.

That said, I would like to see what a user would enter for a remotereadspec with a URL and TLS/Auth configuration.

Reply:
Make sense! We will close the UI PR. So users could just use the advanced options for now.

Reply:
Hey,

we use multiple remote_write urls, basic auth and also the queue_config fields.
I'd be fine with have one field for the whole remote write spec.